lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -36,94 +137,196 @@ class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
                   ConversionPatternRewriter &rewriter) const override {
     if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
       return failure();
-    Location loc = op->getLoc();
+
     Value self = adaptor.self();
-    Value ceilMode = adaptor.ceil_mode();
+    Value maxPool2d;
 
+    SmallVector<int64_t, 2> kernelSizeInts, strideInts, paddingInts,
+        dilationInts;
+    if (failed(checkAndGetPoolingParameters<AtenMaxPool2dOp>(
```

Comment:
for `max_pool2d_with_indices` and `max_pool2d` the input can be 3D. We need to guard the 3D case if we don't want to support that.